moulinette/core.py

```diff
@@ -573,7 +573,7 @@ def acquire(self):
                     self._lock()
                     break
 
-            if (time.time() - start_time) > self.timeout:
+            if self.timeout is not None and (time.time() - start_time) > self.timeout:
```

Comment:
Now that I understood the meaning of life, please disregard my remarks about the timeout = 0 case